android/src/main/java/com/rabinapps/native_drag_n_drop/NativeDropView.java

```diff
@@ -0,0 +1,384 @@
+                case DragEvent.ACTION_DRAG_EXITED:
+                    // Ignore events
+                    return true;
+                // An unknown action type was received.
+                default:
+                    Log.w("[DART/NATIVE]","NativeDropView.viewDragListener: Unknown action type received by View.OnDragListener.");
+                    break;
+            }
+            return false;
+        };
+    }
+
+    private void showToast(String message) {
+        Toast.makeText(context, message, Toast.LENGTH_LONG).show();
+    }
+
+    private boolean isImage(String mime) {
+        return mime.startsWith("image/");
+    }
+
+    private boolean isText(String mime) {
+        return mime.startsWith("text/");
+    }
+
+    private boolean isPdf(String mime) {
+        return mime.startsWith("application/pdf");
+    }
+
+    private boolean isVideo(String mime) {
+        return mime.startsWith("video/");
+    }
+
+    private boolean isAudio(String mime) {
+        return mime.startsWith("audio/");
+    }
+    private boolean isUri(String mime){return mime.startsWith("text/uri");}
+
+    private boolean shouldAllowAudio(){
+        return this.allowedDropDataTypes.contains("audio");
+    }
+    private boolean shouldAllowImages(){
+        return this.allowedDropDataTypes.contains("image");
+    }
+    private boolean shouldAllowVideos(){
+        return this.allowedDropDataTypes.contains("video");
+    }
+    private boolean shouldAllowPdfs(){
+        return this.allowedDropDataTypes.contains("pdf");
+    }
+    private boolean shouldAllowFiles(){
+        return this.allowedDropDataTypes.contains("file");
+    }
+    private boolean shouldAllowText(){
+        return this.allowedDropDataTypes.contains("text");
+    }
+    private boolean shouldAllowUrl(){
+        return this.allowedDropDataTypes.contains("url");
+    }
+    private boolean isFileAllowed(Uri uri){
+        String extension = getFileExtension(activity, uri).substring(1);
+        return this.allowedDropFileExtensions.contains(extension);
+    }
+
+    private void handleDroppedData(DragEvent event){
+        final ArrayList<Map<String, Object>> data = new ArrayList<>();
+
+        View vw = (View) event.getLocalState();
+        if (vw != null){
+            sendDropData(data);
+            return;
+        }
+        ClipData clipData = event.getClipData();
+        int clipCount = clipData.getItemCount();
+
+        new Thread(() -> {
+            for (int i = 0; i <clipCount; i++){
+
+                ClipData.Item item = clipData.getItemAt(i);
+                String mimeType = "text/plain";
+                DragAndDropPermissionsCompat dropPermissions = null;
+                if (item.getUri() != null && ContentResolver.SCHEME_CONTENT.equals(item.getUri().getScheme())) {
+                    // Accessing a "content" scheme Uri requires a permission grant.
+
+                    dropPermissions = ActivityCompat
```

Comment:
Check if dropPermissions == null. If it is log that `"Permission was denied, ignoring file."` and `continue` to next item in loop

If it needs permission and it can't get it, it will crash, so it is important to catch that.